packages/nhost-js/src/utils/helpers.ts

```diff
@@ -25,16 +27,30 @@ export function urlFromSubdomain(
     throw new Error('Either `backendUrl` or `subdomain` must be set.')
   }
 
-  // check if subdomain is [http[s]://]localhost[:port]
+  // check if subdomain is [http[s]://]localhost[:port] or [http[s]://]local[:port]
   const subdomainLocalhostFound = subdomain.match(LOCALHOST_REGEX)
   if (subdomainLocalhostFound?.groups) {
-    const { protocol = 'http', host, port = 1337 } = subdomainLocalhostFound.groups
+    const { protocol, host, port } = subdomainLocalhostFound.groups
 
     const urlFromEnv = getValueFromEnv(service)
     if (urlFromEnv) {
       return urlFromEnv
     }
-    return `${protocol}://${host}:${port}/v1/${service}`
+
+    if (host === 'localhost') {
+      console.warn(
+        'The `subdomain` is set to "localhost". Support for this will be removed in a future release. Please use "local" instead.'
+      )
+
+      return `${protocol || 'http'}://localhost:${port || 1337}/v1/${service}`
+    }
+
+    // TODO: Change `localdev` to `local`
```

Comment:
what is the reason for not changing this already?

Reply:
I mentioned this at one of our meetings. For now, I wanted to keep it `localdev`, so it can be tested with the existing CLI. I'll change it to `local` everywhere eventually.